### PR TITLE
Add Copilot agents, instructions, testing guidelines, and dependency analysis skill

### PR DESCRIPTION
**Pull request by dpaulson45**
## Summary

Adds repository-level Copilot configuration for CSS-Exchange to standardize AI-assisted development workflows.

### Agents (.github/agents/)
- **code-creator** — Creates/modifies PowerShell scripts with repo quality validation via `Invoke-CodeFormatterOnFiles`
- **code-review** — Multi-model PR review with scoped and scope-blind analysis passes (4-pass grid)
- **pester-test** — Creates Pester tests following repo patterns and performance rules
- **pr-review-loop** — Iterative PR review remediation with circuit breakers

### Instructions and Guidelines
- **copilot-instructions.md** — Behavioral rules, code quality standards, architecture overview, personal override workflow
- **pester-testing-guidelines.md** — Comprehensive Pester testing cookbook with examples, mocking patterns, coverage requirements
- **Example scripts** — Three script/test pairs demonstrating getter, Exchange cmdlet mocking, and state-changing patterns

### Skills (.github/skills/)
- **dependency-analysis** — Maps cascading impact of shared function changes via build system XML (BFS traversal)

### Build System
- Exclude `.github/` from Build.ps1 and Pester.ps1 discovery
- Add cspell dictionary entry for `github`

### Infrastructure
- Gitignore personal override directories (`.copilot/`, `.claude/`, `.agents/`)
- Document merge vs replace semantics for personal overrides

### Key Design Decisions
- Agents use `Invoke-CodeFormatterOnFiles` for targeted file formatting; `CodeFormatter.ps1` documented as convenience wrapper
- Only code-modifying agents (code-creator, pester-test, pr-review-loop) use `-Save` flag; code-review does not
- Personal instructions merge additively; personal agent overrides replace entirely (documented with warnings)

### CODE REVIEW

File: .github/examples/Get-ServerConfig.ps1 [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            ServerName        = $ServerName
+            IsOnline          = $true
+            ConfigVersion     = "1.0"
+            LastUpdated       = (Get-Date)
+            ProcessorCount    = 4
+            AvailableMemoryGB = 16
+        }
+    } catch {
+        Write-Error "Failed to retrieve config for $ServerName : $_"
+        throw
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Get-ServerConfig.ps1]: Fixed — refactored from script-level params to a unction Get-ServerConfig {} definition so dot-sourcing loads the function without executing it.

File: .github/examples/Get-ServerConfig.ps1 [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    [string]$ServerName,
+
+    [Parameter()]
+    [ValidateRange(1, 300)]
+    [int]$Timeout = 30
+)
+
+begin {
+    $Script:ErrorActionPreference = "Stop"
+    Write-Verbose "Starting Get-ServerConfig for server: $ServerName with timeout: $Timeout seconds"
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Get-ServerConfig.ps1]: Fixed — changed to $ErrorActionPreference = 'Stop' (no $Script: prefix).

File: .github/examples/Update-ServerSetting.ps1 [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    [ValidateNotNullOrEmpty()]
+    [string]$ConfigValue
+)
+
+begin {
+    $Script:ErrorActionPreference = "Stop"
+    $validKeys = @("MaxMemory", "MaxCPU", "HealthCheckInterval", "LogRetention")
+}
+
+process {
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Update-ServerSetting.ps1]: Fixed — refactored from script-level params to a unction Update-ServerSetting {} definition.

File: .github/examples/Update-ServerSetting.ps1 [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    [string]$ConfigKey,
+
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [string]$ConfigValue
+)
+
+begin {
+    $Script:ErrorActionPreference = "Stop"
+    $validKeys = @("MaxMemory", "MaxCPU", "HealthCheckInterval", "LogRetention")
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Update-ServerSetting.ps1]: Fixed — changed to $ErrorActionPreference = 'Stop' (no $Script: prefix).

File: .github/examples/Test-ServerHealth.ps1 [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [string]$Identity
+)
+
+begin {
+    $Script:ErrorActionPreference = "Stop"
+}
+
+process {
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Test-ServerHealth.ps1]: Fixed — refactored from script-level params to a unction Test-ServerHealth {} definition.

File: .github/examples/Test-ServerHealth.ps1 [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [string]$Identity
+)
+
+begin {
+    $Script:ErrorActionPreference = "Stop"
+}
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Test-ServerHealth.ps1]: Fixed — changed to $ErrorActionPreference = 'Stop' (no $Script: prefix).

File: .github/examples/Get-ServerConfig.Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+<#
+.SYNOPSIS
+    Example of test pattern for a function that modifies state and handles errors.
+
+.DESCRIPTION
+    This demonstrates:
+    - Testing state-changing operations
+    - Mocking dependencies
+    - Error handling patterns
+    - Validation logic
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Get-ServerConfig.Tests.ps1]: Fixed — updated synopsis to 'getter function with parameter validation' to match the actual content.

File: .github/skills/dependency-analysis/Get-DependencyCascade.ps1 [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        }
+    }
+}
+
+$cascadingOnly = $allDependents | Where-Object { $_ -notin $directCallers }
+
+# Root/released scripts
+$rootScripts = $allDependents | Where-Object {
+    -not $dependentHashtable.ContainsKey($_)
+}
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/skills/dependency-analysis/Get-DependencyCascade.ps1:106]: Fixed — added explicit null and scalar guards for $directCallers, $cascadingOnly, and $rootScripts. Each variable is now checked: if null, set to empty array; if not an array, wrapped in array.

File: .github/pester-testing-guidelines.md [hunk 9]
```diff
@@ -0,0 +1,565 @@
...
+```
+
+Run tests for specific files changed on main branch:
+```powershell
+.build/Pester.ps1 -Branch main
+```
+
+Run specific test file:
+```powershell
+.build/Pester.ps1 -Files "Admin/MonitorExchangeAuthCertificate/DataCollection/Tests/Get-ExchangeAuthCertificateStatus.Tests.ps1"
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/pester-testing-guidelines.md]: Fixed — changed to Invoke-Pester -Path since .build/Pester.ps1 does not have a -Files parameter.

File: .github/skills/dependency-analysis/Get-DependencyCascade.ps1 [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+    Write-Error "Target file not found: $TargetPath"
+    return
+}
+
+# Level 1: Direct callers
+$directCallers = @()
+if ($dependentHashtable.ContainsKey($resolvedTarget)) {
+    $directCallers = $dependentHashtable[$resolvedTarget]
+}
+
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/skills/dependency-analysis/Get-DependencyCascade.ps1:80]: Fixed — when the target has no dependents, the report now shows 'Target is a root/entry-point script with no dependents. Changes affect only this file.' instead of empty sections with zeros.

File: .github/skills/dependency-analysis/Get-DependencyCascade.ps1 [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+    [switch]$BuildFirst
+)
+
+# cspell:ignore toplevel
+$repoRoot = (git rev-parse --show-toplevel 2>$null)
+if ($repoRoot) {
+    $repoRoot = (Resolve-Path $repoRoot).Path
+} else {
+    $repoRoot = (Get-Item "$PSScriptRoot\..\..\..").FullName
+}
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/skills/dependency-analysis/Get-DependencyCascade.ps1:39]: Fixed — wrapped the git rev-parse call in try/catch so CommandNotFoundException is caught and the fallback to $PSScriptRoot\..\..\.. runs when git is unavailable.

File: .github/skills/dependency-analysis/Get-DependencyCascade.ps1 [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+$directCallers | ForEach-Object { $queue.Enqueue($_); $null = $allDependents.Add($_) }
+while ($queue.Count -gt 0) {
+    $current = $queue.Dequeue()
+    if ($dependentHashtable.ContainsKey($current)) {
+        foreach ($dep in $dependentHashtable[$current]) {
+            if ($allDependents.Add($dep)) {
+                $queue.Enqueue($dep)
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/skills/dependency-analysis/Get-DependencyCascade.ps1:96]: Fixed — BFS loop now normalizes each hashtable value: checks for null, and wraps scalars in an array before iterating with foreach.

File: .build/Pester.ps1 [hunk 13]
```diff
@@ -42,7 +42,7 @@ begin {
 
     $root = Get-Item "$PSScriptRoot\.."
     $scripts = @(Get-ChildItem -Recurse $root |
-            Where-Object { $_.Name -like "*.Tests.ps1" })
+            Where-Object { $_.Name -like "*.Tests.ps1" -and $_.FullName -notmatch "\.github" })
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/Pester.ps1:45]: This is pre-existing behavior in Pester.ps1, not introduced by this PR. The only change this PR makes to Pester.ps1 is adding the `.github` exclusion filter. Leaving for the author to address separately if desired.

File: .build/CodeFormatter.ps1 [hunk 14]
```diff
@@ -40,7 +40,7 @@ if ($optimizeCodeFormatter) {
     $filesToCheck.FullName | Write-Host
 } else {
     $filesToCheck = Get-ChildItem -Path $repoRoot -Directory | Where-Object {
-        $_.Name -ne "dist" } | ForEach-Object {
+        $_.Name -ne "dist" -and $_.Name -ne ".github" } | ForEach-Object {
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/CodeFormatter.ps1]: This is pre-existing behavior in CodeFormatter.ps1, not introduced by this PR. The only change this PR makes is adding `.github` to the directory exclusion. Leaving for the author to address separately if desired.

File: .github/examples/README.md [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+## Key Patterns Demonstrated
+
+| Pattern | Location | Reference |
+|---------|----------|-----------|
+| Basic parameter validation | Get-ServerConfig.Tests.ps1 lines 69-87 | "Show this to test null/empty checks" |
+| Valid/invalid combinations | Update-ServerSetting.Tests.ps1 lines 62-81 | "Show this for format validation" |
+| Mocking external cmdlets | Test-ServerHealth.Tests.ps1 lines 26-46 | "Show this to mock Get-ExchangeServer" |
+| Error path testing | Get-ServerConfig.Tests.ps1 lines 89-99 | "Show this for error handling tests" |
+| Edge cases | Get-ServerConfig.Tests.ps1 lines 101-117 | "Show this for boundary testing" |
+| Output validation | Get-ServerConfig.Tests.ps1 lines 119-137 | "Show this for asserting output structure" |
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/README.md]: Fixed — updated all line-number references in the Key Patterns table and example dialogue to match the current files after function refactoring.

File: .github/pester-testing-guidelines.md [hunk 16]
```diff
@@ -0,0 +1,565 @@
...
+    
+    It "Should correctly order certificates by expiration date" {
+        $certs = @(
+            (New-AuthCertificateUnitTestObject -NotAfter '2024-12-31'),
+            (New-AuthCertificateUnitTestObject -NotAfter '2023-06-30'),
+            (New-AuthCertificateUnitTestObject -NotAfter '2025-01-01')
+        )
+        $sorted = $certs | Sort-Object NotAfter
+        $sorted[0].NotAfter | Should -Be '2023-06-30'
+        $sorted[-1].NotAfter | Should -Be '2025-01-01'
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/pester-testing-guidelines.md:318]: This is a documentation example in the testing guidelines, not runnable code. The implicit DateTime comparison is adequate for illustrating the sorting pattern. Leaving for the author.

File: .build/CodeFormatter.ps1 (hunk 14 above)
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/CodeFormatter.ps1]: The `.github` exclusion is intentional — these are agent/documentation files, not production scripts. Contributors modifying `.github/*.ps1` files use `Invoke-CodeFormatterOnFiles` directly on specific files (documented in copilot-instructions.md). Leaving for the author.

File: .github/skills/dependency-analysis/Get-DependencyCascade.ps1 [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+        Write-Error "Build.ps1 failed: $_. Cannot proceed with stale dependency data."
+        return
+    }
+}
+
+$dependentXml = Join-Path $distFolder "dependentHashtable.xml"
+$dependencyXml = Join-Path $distFolder "dependencyHashtable.xml"
+
+if (-not (Test-Path $dependentXml) -or -not (Test-Path $dependencyXml)) {
+    Write-Error "Dependency XML files not found in $distFolder. Run .build/Build.ps1 first."
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/skills/dependency-analysis/Get-DependencyCascade.ps1:56]: The existence check for both XML files is intentional — Build.ps1 generates both files together, so if either is missing it indicates the build hasn't been run. The check serves as a health guard before proceeding. Leaving for the author.

File: .github/examples/Get-ServerConfig.Tests.ps1 [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+            $result.PSObject.Properties.Name | Should -Contain "ProcessorCount"
+        }
+
+        It "Should set ServerName correctly" {
+            $result.ServerName | Should -Be "SERVER01"
+        }
+
+        It "Should have correct data types" {
+            $result.IsOnline | Should -BeOfType [bool]
+            $result.ProcessorCount | Should -BeOfType [int]
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Get-ServerConfig.Tests.ps1]: Fixed. Changed all `$result` references in `It` blocks to `$Script:result` where the variable was assigned in `BeforeAll` with `$Script:result = ...`. Local `$result` assignments within individual `It` blocks (e.g., Edge Cases context) correctly remain as `$result`.

File: .github/examples/Get-ServerConfig.Tests.ps1 [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+            $Script:result = Get-ServerConfig -ServerName "SERVER01"
+        }
+
+        It "Should always set IsOnline to true for valid servers" {
+            $result.IsOnline | Should -Be $true
+        }
+
+        It "Should set LastUpdated to mocked time" {
+            $result.LastUpdated | Should -Be ([DateTime]::Parse('2024-06-15T12:00:00'))
+        }
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Get-ServerConfig.Tests.ps1:117]: Fixed. Same `$Script:result` scoping fix applied to the Output Validation context.

File: .github/examples/Update-ServerSetting.Tests.ps1 [hunk 20]
```diff
@@ -0,0 +1,174 @@
...
+            $result.PSObject.Properties.Name | Should -Contain "Timestamp"
+        }
+
+        It "Should have correct property types" {
+            $result.Updated | Should -BeOfType [bool]
+            $result.Timestamp | Should -BeOfType [DateTime]
+        }
+
+        It "Should set Timestamp to mocked time" {
+            $result.Timestamp | Should -Be ([DateTime]::Parse('2024-06-15T12:00:00'))
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Update-ServerSetting.Tests.ps1]: Fixed. Changed all `$result` references in `It` blocks to `$Script:result` in the Output Validation context where `BeforeAll` assigns `$Script:result`.

File: .github/examples/Test-ServerHealth.Tests.ps1 [hunk 21]
```diff
@@ -0,0 +1,192 @@
...
+            $result.PSObject.Properties.Name | Should -Contain "DaysUntilExpire"
+        }
+
+        It "Should set ServerName from Exchange response" {
+            $result.ServerName | Should -Be "EX01"
+        }
+
+        It "Should report healthy status for valid certificate" {
+            $result.CertificateStatus | Should -Be "Healthy"
+            $result.DaysUntilExpire | Should -BeGreaterOrEqual 364
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Test-ServerHealth.Tests.ps1]: Fixed. Changed all `$result` references in `It` blocks to `$Script:result` in the Happy Path context where `BeforeAll` assigns `$Script:result`.

File: .github/examples/README.md [hunk 22]
```diff
@@ -0,0 +1,86 @@
+# Example Test Scripts Reference
+
+This folder contains minimal example scripts and tests for reference when assisting developers with Pester testing patterns.
+
+**DO NOT** include these in production builds or releases.
+
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/README.md:6]: Fixed. Added a note to README.md explaining that these examples intentionally colocate test files for simplicity, and that production tests should follow the `Tests/` subfolder convention.

File: .github/pester-testing-guidelines.md [hunk 23]
```diff
@@ -0,0 +1,565 @@
...
+- **Location**: Colocated in `Tests/` subdirectory next to the script being tested
+- **Example**:
+  ```
+  Shared/Tests/Get-ExchangeBuildVersionInformation.Tests.ps1
+  Admin/MonitorExchangeAuthCertificate/DataCollection/Tests/Get-ExchangeAuthCertificateStatus.Tests.ps1
+  ```
+
+### Basic Test Structure
+
+Every test file must follow this template:
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/pester-testing-guidelines.md]: Fixed. Changed "Every test file must follow this template" to "Every test file should follow this recommended template (some established tests like HealthChecker may differ in structure)".

File: .build/CodeFormatter.ps1 (hunk 14 above)
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/CodeFormatter.ps1]: Not addressed — the PR author directed not to change `.build/CodeFormatter.ps1` or `.build/Pester.ps1` without a stronger justification. These exclusions would be a separate scope of work if desired.

File: .build/Pester.ps1 (hunk 13 above)
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/Pester.ps1:45]: Not addressed — the PR author directed not to change `.build/CodeFormatter.ps1` or `.build/Pester.ps1` without a stronger justification. These exclusions would be a separate scope of work if desired.

File: .github/skills/dependency-analysis/SKILL.md [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+$directCallers | ForEach-Object { $queue.Enqueue($_); $null = $allDependents.Add($_) }
+while ($queue.Count -gt 0) {
+    $current = $queue.Dequeue()
+    if ($dependentHashtable.ContainsKey($current)) {
+        foreach ($dep in $dependentHashtable[$current]) {
+            if ($allDependents.Add($dep)) {
+                $queue.Enqueue($dep)
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/skills/dependency-analysis/SKILL.md:62]: Acknowledged. Valid observation about PowerShell string enumeration in `foreach`, but this is a documentation example in SKILL.md illustrating the BFS concept, not executable production code. Leaving for the author to decide whether to add `@()` wrapping in the example.

File: .github/examples/Update-ServerSetting.Tests.ps1 [hunk 25]
```diff
@@ -0,0 +1,174 @@
...
+        }
+
+        It "Should set Timestamp to mocked time" {
+            $Script:result.Timestamp | Should -Be ([DateTime]::Parse('2024-06-15T12:00:00'))
+        }
+    }
+
+    Context "Edge Cases" {
+        It "Should handle ConfigKey with mixed case" {
+            $result = Update-ServerSetting -ConfigKey "MaxMemory" -ConfigValue "32GB"
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Update-ServerSetting.Tests.ps1]: Acknowledged. The test name suggests mixed-case testing but passes a normally-cased key. This is original PR code not modified by the automated review fixes. Leaving for the author.

File: .github/examples/Update-ServerSetting.Tests.ps1 [hunk 26]
```diff
@@ -0,0 +1,174 @@
...
+        It "Should provide helpful error message for invalid key" {
+            $errorAction = $null
+            try {
+                Update-ServerSetting -ConfigKey "BadKey" -ConfigValue "value" -ErrorAction Stop
+            } catch {
+                $errorAction = $_
+            }
+
+            $errorAction | Should -Not -BeNullOrEmpty
+        }
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Update-ServerSetting.Tests.ps1:174]: Acknowledged. The test could assert on the error message content for a more meaningful example. This is original PR code not modified by the automated review fixes. Leaving for the author.

File: .build/CodeFormatter.ps1 (hunk 14 above)
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/CodeFormatter.ps1]: [CLI] - This looks like something that should be done. Please remove the `.github` here.

File: .github/examples/Update-ServerSetting.Tests.ps1 (hunk 25 above)
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Update-ServerSetting.Tests.ps1]: [CLI] - Review and determine what this test is trying to do/mock from main repo. Apply if applicable.

File: .build/CodeFormatter.ps1 (hunk 14 above)
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/CodeFormatter.ps1]: Fixed per author direction. Removed the `.github` exclusion from CodeFormatter.ps1 so `.github/**/*.ps1` and `.github/**/*.md` files are now included in full-repo formatting scans.

[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/CodeFormatter.ps1]: Partially addressed. The `.github` exclusion has been removed so CodeFormatter now scans `.github`. The `.copilot/`, `.claude/`, `.agents/` directories are gitignored and would only exist locally — leaving those for the author to decide on separately.

File: .github/examples/Update-ServerSetting.Tests.ps1 (hunk 25 above)
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Update-ServerSetting.Tests.ps1]: Fixed per [CLI] directive. Changed test input from `"MaxMemory"` (exact match) to `"mAxMeMoRy"` (genuinely mixed case) and added `$result.Updated | Should -Be $true` to verify the function accepts and processes it. Also added `$errorAction.Exception.Message | Should -BeLike "*Invalid configuration key*"` to the error message test.

File: .github/examples/Update-ServerSetting.Tests.ps1 [hunk 27]
```diff
@@ -0,0 +1,174 @@
...
+
+        It "Should set Timestamp to mocked time" {
+            $Script:result.Timestamp | Should -Be ([DateTime]::Parse('2024-06-15T12:00:00'))
+        }
+    }
+
+    Context "Edge Cases" {
+        It "Should handle ConfigKey with mixed case" {
+            $result = Update-ServerSetting -ConfigKey "MaxMemory" -ConfigValue "32GB"
+            $result | Should -Not -BeNullOrEmpty
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Update-ServerSetting.Tests.ps1]: Fixed. The test now passes `"mAxMeMoRy"` (genuinely mixed case) and asserts `$result.Updated | Should -Be $true` to verify the function accepts it.

File: .github/examples/Update-ServerSetting.Tests.ps1 (hunk 26 above)
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/examples/Update-ServerSetting.Tests.ps1:174]: Fixed. Added `$errorAction.Exception.Message | Should -BeLike "*Invalid configuration key*"` to assert on the error message content.

File: .build/Build.ps1 [hunk 28]
```diff
@@ -43,7 +43,7 @@ New-Item -Path $distFolder -ItemType Directory | Out-Null
     are referenced.
 #>
 
-$excludedFolders = @(".build", "dist")
+$excludedFolders = @(".build", "dist", ".github")
 
 $allScriptFiles = Get-ChildItem -Path $repoRoot -Directory |
     Where-Object { -not $excludedFolders.Contains($_.Name) } |
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/Build.ps1:49]: Acknowledged. The `.github` exclusion was removed from CodeFormatter.ps1 per author direction, so `.github/` is now included in full-repo scans. The PR description may need updating to reflect this change. Leaving for the author.

[the remark this answers is not in the document]
Comment (dpaulson45) [reply .build/Build.ps1:49]: Fixed. Updated PR description to remove CodeFormatter.ps1 from the `.github` exclusion list. Now reads "Exclude `.github/` from Build.ps1 and Pester.ps1 discovery".

File: .github/copilot-instructions.md [hunk 29]
```diff
@@ -0,0 +1,252 @@
...
+- Dictionary is in `.build/cspell-words.txt`
+- Most spelling issues are caught by using proper camelCase or PascalCase variable names
+
+### Testing (Pester)
+
+```powershell
+.build/Pester.ps1
+```
+
+- Runs all Pester tests (`*.Tests.ps1`) found recursively across the repository
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/copilot-instructions.md]: Fixed. Updated to "Runs all Pester tests (`*.Tests.ps1`) found recursively across the repository (excludes `.github/`)".

File: .github/copilot-instructions.md [hunk 30]
```diff
@@ -0,0 +1,252 @@
...
+
+### Spell Checking
+
+```powershell
+.build/SpellCheck.ps1
+```
+
+- Uses `cspell` (Node.js-based tool)
+- Dictionary is in `.build/cspell-words.txt`
+- Most spelling issues are caught by using proper camelCase or PascalCase variable names
```
[the remark this answers is not in the document]
Comment (dpaulson45) [reply .github/copilot-instructions.md:119]: Fixed. Added note: "`.github/` is excluded from spell checking via `.build/cspell.json`".

**Review by lusassl-msft — APPROVED** [review]
